### ISSUE TITLE
Auto imports don't work on monorepos when using yarn pnp

### ISSUE THREAD

**Issue by Arthur944** [pmizio/typescript-tools.nvim#282]
Problem: When using a monorepo with yarn pnp, the auto import code action never lists options for other packages within the monorepo. Interestingly, after saving the tsconfig.json for one of the sub-packages within the monorepo, auto import works fine for a few seconds, but then it goes back to the same behavior.

Here's a rerpoduction repository: https://github.com/Arthur944/typescript-tools-repro

Video of the issue: 

https://github.com/pmizio/typescript-tools.nvim/assets/26466294/a5f4006b-b77a-447d-bc4b-7d73789c72bd

(This happened on an M3 Macbook)

**Comment by Arthur944** [comment]
Never mind, I just needed an "include": ["src", "../common/src"] in my packages/frontend/tsconfig.json

**Comment by Arthur944** [comment]
Actually, my original reproduction was incorrect, but the issue does exist. The current state of the reproduction repository shows the auto imports not working, and the branch "working-with-bun" shows the same code, just with yarn replaced with bun. On that branch auto imports work correctly.

**Comment by Arthur944** [comment]
Looks like it might be an issue in yarn itself https://github.com/yarnpkg/berry/issues/3999

**Comment by melroy89** [comment]
> Never mind, I just needed an "include": ["src", "../common/src"] in my packages/frontend/tsconfig.json

yeah so actually this is a bad pratice. Since normally you only put `"src/**/*.d.ts"` to the include. And under `"files": ` you only need to specify the entry point (eg. your `main.ts`). .. `tsc` will then work fine. Also no potentially warnings that might say:

> Warning: somefile.ts  is part of the TypeScript compilation but it's unused. Add only entry points to the 'files' or 'include' properties in your tsconfig.

I currently use this:

```json
  "compilerOptions": {
    "outDir": "./out-tsc/app",
    "types": [],
    "paths": {
      "@app/*": ["src/app/*"],
      "@components/*": ["src/app/components/*"],
      "@environments/*": ["src/environments/*"],
      "@interfaces/*": ["src/app/interfaces/*"]
    }
  },
 "files": ["src/main.ts", "src/polyfills.ts"],
 "include": ["src/**/*.d.ts"]
```

But its not working in the editor.. It has 100% something to do with the TS language server that is just too stupid.

Compiling goes fine btw.
